yt_dlp/extractor/wistia.py

```diff
@@ -39,6 +40,21 @@ def _extract_media(self, embed_config):
         video_id = data['hashedId']
         title = data['name']
 
+        def get_real_ext(url):
+            res = self._request_webpage(HEADRequest(url), None, note=False, errnote=False)
+            if res:
+                meta_ext = res.headers.get('x-amz-meta-name', '').rpartition('.')[2]
+                if meta_ext:
+                    return meta_ext
+                content_type = res.headers.get('content-type')
+                if content_type in ('video/mp4', 'video/quicktime'):
+                    return 'mp4'
+                elif content_type == 'image/png':
+                    return 'png'
+                elif content_type == 'image/jpeg':
+                    return 'jpg'
+            return 'bin'
```

Comment:
move outside closure imo. It may even be moved o common

Reply:
> It may even be moved to common

there are a couple of similar functions in utils, `urlhandle_detect_ext` and `mimetype2ext`, but:
 - `urlhandle_detect_ext` doesn't allow for a default extension to be returned
 - `mimetype2ext` doesn't know of any image types
 - Wistia (wrongly) uses `video/quicktime` for mp4 containers (confirmed w/ffprobe) and `mimetype2ext` would return `mov`

so I decided to write this Wistia-specific function instead.

However, I do think that `mimetype2ext` could be improved upon so that it could be applicable in this case (and more cases in general), if you think we should do that

Reply:
> mimetype2ext doesn't know of any image types
> However, I do think that mimetype2ext could be improved upon so that it could be applicable in this case (and more cases in general), if you think we should do that

That sounds good. You can also add `x-amz-meta-name` to `urlhandle_detect_ext`

If that is done, the code will become like this, correct?
```py
        def get_real_ext(url):
            urlh = self._request_webpage(...)
            if urlh:
                ext = urlhandle_detect_ext(urlh)
            return 'mp4' if ext == 'mov' else ext or 'bin'
```

Reply:
Does this look OK? https://github.com/yt-dlp/yt-dlp/pull/5415/commits/3fdaf69460a3b9195073ac5bcdc8884714df4b2b